rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala

```diff
@@ -141,6 +141,34 @@ object Exponential {
     standard.scale(Real.one / rate)
 }
 
+/**
+  * A Beta distribution with expectation `a/(a + b)` and variance `ab/((a + b)^2 (1 + a + b))`.
+  */
+object Beta {
+  def apply(a: Real, b: Real): Continuous = new Continuous {
+    def realLogDensity(p: Real): Real =
+      If(p < 0,
+         Real.zero.log,
+         If(p > 1,
+            Real.zero.log,
+            (a - 1) * p.log + (b - 1) * (1 - p).log - Combinatrics.beta(a, b)))
+
+    def param: RandomVariable[Real] = {
+      val x = new Variable
+      val logistic = Real.one / (Real.one + (x * -1).exp)
+      val logisticJacobian = logistic * (1 - logistic)
+      val density = realLogDensity(logistic) + logisticJacobian.log
+      RandomVariable(logistic, density)
+    }
+
+    def generator: Generator[Double] = Generator.require(Set(a, b)) { (r, n) =>
+      val z1 = Gamma(a, 1).generator.get(r, n)
```

Comment:
it's slightly wasteful to create `Gamma(a,1).generator` (and the one for b) for each sample. You can pull these out of the closure and into temps in the `generator` method body.

Reply:
In fact, you can do this more cleanly with a `zip` and `map` on those two generator objects.